### DISCUSSION TITLE
Unable to get Web Search Working

### DISCUSSION THREAD

**Discussion by Prime-AI28** [open-webui/open-webui#4761] [General]
# Bug Report

Getting the following error when trying to use web search feature:

requests.exceptions.SSLError: HTTPSConnectionPool(host='host.docker.internal', port=8080): Max retries exceeded with url: /search?q=hello%2C+can+you+tell+me+about+today%27s+fresh+new+from+times+of+India%3F&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)')))

## Installation Method

i have used docker latest image to install open-webui, then i have followed the web search documentation by, making a docker container for searxng using the latest image.

## Environment

- **Open WebUI Version:** v0.3.13
- **Ollama:** v0.36
- **Operating System:**  Windows 11

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Web search function should use searxng hosted on the docker to search and scrape the web page and give them to the llm.

## Actual Behavior:

unable to make a successful web search due to ssl Error, the search query that is generated is not correct/valid because even if i manual try to open that web url it shows 403 Forbidden. 

## Description

**Bug Summary:**Unable to get the web search function get working, despite the searxng docker working completely fine when trying to access it using open-webui, it isn't able to get result, tried both: bypass SLL on and off.

## Reproduction Details

**Steps to Reproduce:**
deploy open-webui v.0.3.13 using the docker image, 
deploy the searxng using the docker image of the latest version

in open-webui -> settings -> admin settings -> web search:

enable web search, choose searxng as the search engine, 

searxng query url : https://host.docker.internal:8080/search?q=<query>

Bypass SSL verification for Websites : on and off

now use any llm model and turn on web search using the + button.

## Logs and Screenshots

**Docker Container Logs:**

ERROR [apps.rag.main] HTTPSConnectionPool(host='host.docker.internal', port=8080): Max retries exceeded with url: /search?q=hello%2C+can+you+tell+me+about+today%27s+fresh+new+from+times+of+India%3F&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)')))
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 466, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 652, in connect
    sock_and_verified = _ssl_wrap_socket_and_match_hostname(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 805, in _ssl_wrap_socket_and_match_hostname
    ssl_sock = ssl_wrap_socket(
               ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/ssl_.py", line 465, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/ssl_.py", line 509, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ssl.py", line 517, in wrap_socket
    return self.sslsocket_class._create(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ssl.py", line 1104, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.11/ssl.py", line 1382, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: [SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 789, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 490, in _make_request
    raise new_e
urllib3.exceptions.SSLError: [SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='host.docker.internal', port=8080): Max retries exceeded with url: /search?q=hello%2C+can+you+tell+me+about+today%27s+fresh+new+from+times+of+India%3F&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/backend/apps/rag/main.py", line 898, in store_web_search
    web_results = search_web(
                  ^^^^^^^^^^^
  File "/app/backend/apps/rag/main.py", line 806, in search_web
    return search_searxng(
           ^^^^^^^^^^^^^^^
  File "/app/backend/apps/rag/search/searxng.py", line 68, in search_searxng
    response = requests.get(
               ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 698, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='host.docker.internal', port=8080): Max retries exceeded with url: /search?q=hello%2C+can+you+tell+me+about+today%27s+fresh+new+from+times+of+India%3F&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)')))
HTTPSConnectionPool(host='host.docker.internal', port=8080): Max retries exceeded with url: /search?q=hello%2C+can+you+tell+me+about+today%27s+fresh+new+from+times+of+India%3F&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)')))

**Screenshots/Screen Recordings (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]
![image](https://github.com/user-attachments/assets/a1c33228-7920-48db-9ca1-01cf7447985a)

![image](https://github.com/user-attachments/assets/80c240dd-68a3-4572-a931-d040d7f6cb4d)

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by daddeldash** [comment]
Found out, that OpenwebUI adds "format=json" to the URL which, for me, ends up in this forbidden message. I don't know if I miss something, but removing it brought me the results

**Comment by CD1010** [comment]
I had opposite. The yaml formats tag did not add json.  I added it and it works, but intermittently.  Yes, it is broken.

**Comment by arjunkrishna** [comment]
after installing searxng docker image, please update the settings.yml in its associated volume. 

add - json under formats

![image](https://github.com/user-attachments/assets/7b17f22e-5095-4e11-a39a-52fbc0e6971f)


https://github.com/open-webui/open-webui/discussions/3003#discussioncomment-9741643

**Reply by Urvesh71** [reply]
Create `settings.yml` file and mount it in volume of web search container
## settings.yml
## Reference: https://docs.searxng.org/admin/settings/settings.html#settings-use-default-settings

```
use_default_settings: true

server:
  secret_key: "YOUR SECRET KEY"
  limiter: false
  image_proxy: true
  port: 8080
  bind_address: "0.0.0.0"

ui:
  static_use_hash: true

search:
  safe_search: 0
  autocomplete: ""
  default_lang: ""
  formats:
    - html
    - json
```

**Reply by mekler22** [reply]
Tried doing all this. settings.yml reverts to its original settings. and web search does not work regardless of engine or settings. it keeps throwing error searching

**Comment by logan683** [comment]
This is a persistent error I am also dealing with.  I have also tried everything posted that can be found.

**Comment by Greg-2600** [comment]
also same issue.
